csrc/ir/container.cpp

```diff
@@ -235,7 +235,7 @@ bool IrContainer::inContainer(const Statement* stmt) const {
 // Shortcuts for frequently used vals
 Val* IrContainer::zeroVal() {
   if (!zero_val_) {
-    auto zero_val = IrBuilder::create<Val>(this, 0L);
+    auto zero_val = IrBuilder::create<Val>(this, 0L, DataType::Index);
```

Comment:
Wonder if all the uses of `zeroVal` are indeed for indexing. Could this be just `int32` and let nvrtc promote to `int64` as needed?

Reply:
I think it makes sense. Will try it.

Reply:
Changed, waiting for CI

Reply:
Looks like I can not change it to `DataType::Int32`, because a great portion for the `zeroVal` usage is for the start of an `IterDomain` which must be index type. I changed this back to index type, and revisited our codebase and changed the `zeroVal()` calls into `zeroVal(dtype)` whereever I think we should.

Reply:
If you changed `zeroVal()` to `zeroVal(dtype)`, doesn't that mean the default type of `zeroVal` can be `int32`?

Reply:
I made the default of `zeroVal` `DataType::Index`, and other dtypes are opt-in. I think this makes sense because `DataType::Index` is the most widely used dtype in our system, much more used than any other dtype.

Reply:
Makes sense.